src/common/utils/printRoot.js

```diff
@@ -51,6 +51,9 @@ function printRoot(root: Collection): string {
     output += '\n';
   }
 
+  // Remove spurious newline at the end.
+  output = output.replace(/\n\n$/, '\n');
```

Comment:
No clue how to get rid of the duplicate line (which we usually want cause the program body comes after).